crates/sozo/ops/src/migration/mod.rs

```diff
@@ -411,68 +420,54 @@ where
         Some(world) => {
             ui.print_header("# World");
 
-            let calldata = vec![strategy.base.as_ref().unwrap().diff.local];
-            let deploy_result =
-                deploy_contract(world, "world", calldata.clone(), migrator, &ui, &txn_config)
-                    .await
-                    .map_err(|e| {
-                        ui.verbose(format!("{e:?}"));
-                        anyhow!("Failed to deploy world: {e}")
-                    })?;
-
-            (world_tx_hash, world_block_number) =
-                if let ContractDeploymentOutput::Output(deploy_result) = deploy_result {
-                    (Some(deploy_result.transaction_hash), deploy_result.block_number)
-                } else {
-                    (None, None)
-                };
-
-            ui.print_sub(format!("Contract address: {:#x}", world.contract_address));
-
-            let offline = ws.config().offline();
+            if world.diff.local_class_hash != world.diff.original_class_hash
+                && world.diff.original_class_hash
+                    == world.diff.remote_class_hash.unwrap_or_default()
```

Comment:
Not sure about this condition and it's related to my previous comment.

To here, we want to actually upgrade anytime local class hash is different from the remote one, independently of the original class hash match.